api/types/access_request.go

```diff
@@ -776,6 +781,28 @@ func (f *AccessRequestFilter) FromMap(m map[string]string) error {
 
 // Match checks if a given access request matches this filter.
 func (f *AccessRequestFilter) Match(req AccessRequest) bool {
+	// only return if the request was made by the api requester
+	if f.Scope == AccessRequestScope_MY_REQUESTS && req.GetUser() != f.Requester {
+		return false
+	}
+	// a user cannot review their own requests
+	if f.Scope == AccessRequestScope_NEEDS_REVIEW && req.GetUser() == f.Requester {
+		return false
+	}
+	// only match if the api requester has submit a review
+	if f.Scope == AccessRequestScope_REVIEWED {
+		reviews := req.GetReviews()
+		var reviewers []string
+		for _, review := range reviews {
+			reviewers = append(reviewers, review.Author)
+		}
+		if !slices.Contains(reviewers, f.Requester) {
+			return false
+		}
+	}
```

Comment:
can we fail faster here by checking if state is not pending and the request is not their own (since you can't review your own request)?

Reply:
sure ill change the main conditional to `if f.Scope == AccessRequestScope_REVIEWED && req.GetUser() != f.Requster`